pkg/kubelet/client/kubelet_client.go

```diff
@@ -98,24 +105,24 @@ func NewStaticKubeletClient(config *KubeletClientConfig) (KubeletClient, error)
 }
 
 // In default HTTPKubeletClient ctx is unused.
-func (c *HTTPKubeletClient) GetConnectionInfo(ctx api.Context, nodeName string) (string, uint, http.RoundTripper, error) {
+func (c *HTTPKubeletClient) GetConnectionInfo(ctx api.Context, nodeName string) (*ConnectionInfo, error) {
 	if errs := validation.ValidateNodeName(nodeName, false); len(errs) != 0 {
-		return "", 0, nil, fmt.Errorf("invalid node name: %s", strings.Join(errs, ";"))
+		return nil, fmt.Errorf("invalid node name: %s", strings.Join(errs, ";"))
 	}
 	scheme := "http"
 	if c.Config.EnableHttps {
 		scheme = "https"
 	}
-	return scheme, c.Config.Port, c.Client.Transport, nil
+	return &ConnectionInfo{scheme, nodeName, c.Config.Port, c.Client.Transport}, nil
```

Comment:
use field names in initialization (e.g. `&ConnectionInfo{Scheme:scheme, Hostname: nodeName, ...}`